.github/workflows/release.yml

```diff
@@ -0,0 +1,70 @@
+name: Release on GitHub & PyPI
+
+on:
+  push:
+    tags:
+      - "v*"
```

Comment:
On release, no?

Reply:
i think it is fine like this. from my point of view, we want to start this pipeline here by just doing a tag with `vX.X.X`, which then builds the wheel and does the release, check the end of the workflow. 
When this workflow creates the release, it will trigger the gh-pages workflow which is on release. if you have another concept in mind let me know.

Reply:
I think we need two steps, when merging a PR, we should check that the wheel creation works.
When releasing a new version, we should publish it.

Reply:
What do you think?

Reply:
ahhh, okay. I see your point. so the first part of this pipeline should run all the time ? or just for merging a PR?
And then the lower part should just run if we put a tag, to signal we want to release. this triggers the upper part, plus the lower part that actually creates the github release and uploads the wheels to pypip.

Reply:
Maybe all the time, but it is quite expansive.
Yes for the second part.

Reply:
mmm, I am wondering... at the end building the wheel using cmeel (https://github.com/cmake-wheel/cmeel) boils down to using cmake with the same setup as in our ci-linux-osx-windows. so maybe it is enough to run it for a PR to make sure, if cmeel changes etc but I don't see how a commit could break the wheels but not the other pipeline.

Reply:
Fine for me.

Reply:
Ok, I did the changes in this way now. Pipeline first part (building wheels) should run on every PR + release, second part (uploading) should just be for release in our repo (done the same way as  https://github.com/robotology/idyntree/blob/master/.github/workflows/python.yml)